test/spec_methodoverride.rb

```diff
@@ -23,6 +23,19 @@ def app
     env["REQUEST_METHOD"].should.equal "PUT"
   end
 
+  should "set rack.errors for invalid UTF8 _method values" do
+    errors = StringIO.new
+    env = Rack::MockRequest.env_for("/",
+      :method => "POST",
+      :input => "_method=\xBF",
+      "rack.errors" => errors)
+
+    app.call env
+
+    errors.rewind
+    errors.read.should =~ /Invalid string for method/
```

Comment:
How come we're using a match here instead of `should.equal`? Can it be something else? I think this is also making the Ruby 1.8.7 build cranky https://travis-ci.org/rack/rack/jobs/365845664#L616

Reply:
I was copying the existing style from a little further down the test:
https://github.com/mclark/rack/blob/a795740a6e071333ecc7b498fdeed24d58750da4/test/spec_methodoverride.rb#L100

I can update both of them if you like?

Reply:
I think the 1.8.7 build is probably failing because I don't think that version of ruby even supports UTF-8, so the error can't be thrown. Does rack have a standard pattern for skipping tests based on the ruby version? I suspect that might be the best solution in that case, but I'm far from an encoding expert (nor do I desire to become one! 😄 )

Reply:
Here's an example https://github.com/rack/rack/blob/1-6-stable/test/spec_chunked.rb#L45

But I prefer wrapping the entire test in the conditional rather than the conditional after the end like:

```
if RUBY_VERSION >= "1.9"
 test blah blah
end
```